netx/httptransport/http3transport_test.go

```diff
@@ -20,6 +21,97 @@ func (d MockHTTP3Dialer) Dial(network, host string, tlsCfg *tls.Config, cfg *qui
 	return quic.DialAddrEarly(host, tlsCfg, cfg)
 }
 
+type MockSNIHTTP3Dialer struct {
+	namech chan string
+}
+
+func (d MockSNIHTTP3Dialer) Dial(network, host string, tlsCfg *tls.Config, cfg *quic.Config) (quic.EarlySession, error) {
+	d.namech <- tlsCfg.ServerName
+	close(d.namech)
+	return quic.DialAddrEarly(host, tlsCfg, cfg)
+}
+
+type MockCertHTTP3Dialer struct {
+	certch chan *x509.CertPool
+}
+
+func (d MockCertHTTP3Dialer) Dial(network, host string, tlsCfg *tls.Config, cfg *quic.Config) (quic.EarlySession, error) {
+	d.certch <- tlsCfg.RootCAs
+	close(d.certch)
+	return quic.DialAddrEarly(host, tlsCfg, cfg)
+}
+
+func TestUnitHTTP3TransportSNI(t *testing.T) {
```

Comment:
```suggestion
func TestHTTP3TransportSNI(t *testing.T) {
```

I suggest dropping `Unit` because this is not a unit test: it's hitting the network. Also, adding the `Unit` and `Integration` words to test names is a convention that I should not have adopted, because tests are poised to change and may start Unit and become Integration over time, and typically the name is not updated. I have recently removed all (or maybe just most) the instances of Unit and Integration from the tree, and I would like the tree in general to discontinue this convention that was a bit difficult to consistently keep.